### ISSUE TITLE
Intermittent crash on startup

### ISSUE THREAD

**Issue by frymaster** [mumble-voip/mumble#1611]
Console log, including first line of next startup

```
<W>2015-03-23 19:58:08.696 DirectSound: Disabled as WASAPI is available
<W>2015-03-23 19:58:08.696 ASIO: No valid devices found, disabling
<W>2015-03-23 19:58:08.727 G15LCDEngine_lglcd: Logitech LCD Manager not detected.
<W>2015-03-23 19:58:08.743 CELT bitstream 8000000b from C:/Program Files (x86)/Mumble/celt0.0.7.0.sse2.dll
<W>2015-03-23 19:58:08.743 CELT bitstream 80000010 from C:/Program Files (x86)/Mumble/celt0.0.11.0.sse2.dll
<W>2015-03-23 19:58:08.743 Locale is "en_GB" (System: "en_GB")
<W>2015-03-23 19:58:08.853 Database SQLite: "3.8.6"
<W>2015-03-23 19:58:08.853 Bonjour: Failed to load dnssd.dll
<W>2015-03-23 19:58:08.853 Overlay: Listening on "\\.\pipe\MumbleOverlayPipe"
<W>2015-03-23 19:58:08.869 Updating application palette
<W>2015-03-23 19:58:08.979 GlobalShortcutWin: Found LowLevelHooksTimeout with value = 0x1388
<W>2015-03-23 19:58:09.136 Adding device {6f1d2b60-d5a0-11cf-bfc7-444553540000} Mouse Mouse:5
<W>2015-03-23 19:58:09.136 Adding device {6f1d2b61-d5a0-11cf-bfc7-444553540000} Keyboard Keyboard:126
<W>2015-03-23 19:58:20.237 DirectSound: Disabled as WASAPI is available
```

Bug report was uploaded

This has happened intermittently for years, however it's only in about the past 9 months that I have been using windows 8, and thus haven't needed to use Ultramon.  I didn't report it as I felt the ultramon injected dll could be interfering.

**Comment by hacst** [comment]
Hi. I found your report ("2015-03-24T19:16 1.3.0 1.3.0~567~g3223c8a~snapshot Unknown.html" for future reference).

Here some preliminary information from the dump:

Crash reason: Unhandled exception at 0x626F7BFA (dinput8.dll) in Dump-5511b820-Win 6.3.9600.1-35585df753e940501e3616844083e9470983a91f.dmp: 0xC0000005: Access violation reading location 0x03536070.

Traceback:
```
 	dinput8.dll!CHid_MungeNotKeyboard()	Unknown
 	dinput8.dll!CHid_InitObjects()	Unknown
 	dinput8.dll!CHid_Init()	Unknown
 	dinput8.dll!CHid_New()	Unknown
 	dinput8.dll!CDIDev_Initialize()	Unknown
 	dinput8.dll!CDIObj_CreateDeviceW()	Unknown
>	mumble_app.dll!GlobalShortcutWin::EnumDevicesCB(const DIDEVICEINSTANCEW * pdidi=0x0320a708, void * pContext=0x02b5abe8) Line 452	C++
```

Corresponding call in GlobalShortcut_win.cpp (L452) EnumDevicesCB:
```cpp
	if (FAILED(hr = cbgsw->pDI->CreateDevice(pdidi->guidInstance, &id->pDID, NULL)))
		qFatal("GlobalShortcutWin: CreateDevice: %lx", hr);
```

https://msdn.microsoft.com/en-us/library/windows/desktop/microsoft.directx_sdk.idirectinput8.idirectinput8.createdevice(v=vs.85).aspx

```
pdidi	0x0320a708 {dwSize=1100 guidInstance={58F3ACE0-3C23-11DE-8002-444553540000} guidProduct={0724045E-0000-0000-0000-504944564944} ...}	const DIDEVICEINSTANCEW *
		dwSize	1100	unsigned long
+		guidInstance	{58F3ACE0-3C23-11DE-8002-444553540000}	_GUID
+		guidProduct	{0724045E-0000-0000-0000-504944564944}	_GUID
		dwDevType	65553	unsigned long
+		tszInstanceName	0x0320a730 L"Microsoft Hardware USB Mouse"	wchar_t[260]
+		tszProductName	0x0320a938 L"Microsoft Hardware USB Mouse"	wchar_t[260]
+		guidFFDriver	{00000000-0000-0000-0000-000000000000}	_GUID
		wUsagePage	12	unsigned short
		wUsage	1	unsigned short
```

Pretty strange for it to crash there. From what I can see our input looks fine.

**Comment by frymaster** [comment]
there have been a few crashes from that email address, are they all showing the a similar trace?

**Comment by hacst** [comment]
About the GUID in question (from http://pastebin.127001.org/1586):
> <W>2010-02-23 12:38:00.125 Adding device {58f3ace0-3c23-11de-8002-444553540000} SideWinder? Mouse SideWinder? Mouse:1025

**Comment by mkrautz** [comment]
@frymaster 

Can you confirm that the crashes only happen with the Sidewinder flightstick plugged in?

Also, does the stick have a pointing device on it?  That would be the mouse identified by DirectInput in your old pastebin.

And, most importantly, does it have mouse buttons?

If not, it might make sense to just blacklist the mouse, unless we can figure out what's actually going on in DInput to cause the crash.

**Comment by frymaster** [comment]
>Also, does the stick have a pointing device on it? That would be the mouse identified by DirectInput in your old pastebin

* Is it not more likely to be my actual mouse? That's also a [sidewinder](http://www.amazon.co.uk/Microsoft-HKA-00003-Sidewinder-Mouse/dp/B000TSLLG0)
* The stick doesn't have a pointing device

**Comment by mkrautz** [comment]
OK, I wasn't sure. The device IDs were so identical that I assumed they had to be from the same device:

<W>2010-02-23 12:38:00.088 Adding device {58eef1f0-3c23-11de-8001-444553540000} SideWinder Force Feedback 2 Joystick SideWinder Force Feedback 2 Joystick:8
<W>2010-02-23 12:38:00.125 Adding device {58f3ace0-3c23-11de-8002-444553540000} SideWinder? Mouse SideWinder? Mouse:1025

(Notice the GUIDs)

...Is the SideWinder(TM) Mouse still in your device logs -- and that's your primary mouse?

In the log in the first post of the issue, it only had:

<W>2015-03-23 19:58:09.136 Adding device {6f1d2b60-d5a0-11cf-bfc7-444553540000} Mouse Mouse:5
<W>2015-03-23 19:58:09.136 Adding device {6f1d2b61-d5a0-11cf-bfc7-444553540000} Keyboard Keyboard:126

That's why I am confused...

**Comment by frymaster** [comment]
>In the log in the first post of the issue, it only had:
Might that be truncated because of the crash?

That being said, I am very confused as well.  I was reminded by this that when I reinstalled win10 after the big update recently, it hadn't reinstalled my mouse software ("Microsoft Mouse and Keyboard Center") so I reinstalled that, and now it's not showing up with "sidewinder" in the name any more.  I dug out the power lead for my joystick and plugged that in, for contrast

https://gist.github.com/frymaster/9ea71af24c155551f00f

tl;dr the joystick doesn't have a pointing device, the mouse was the one identified by "sidewinder?" but it isn't always

I also don't seem to have crashed much on this install - I have an empty dump file from 18/11/2015 but nothing since - so not sure what's going on.  Any time this happens and there's not good reason to suspect other issues, I am keeping uploading crash dumps with the text "crashed on startup"

**Comment by Krzmbrzl** [comment]
As no further crashs are reported, I'll assume the issue to be gone for now
